### PR TITLE
[ new ] add Windows support

### PR DESCRIPTION
**Pull request by elseLuna**
This is an implementation of the changes I proposed in #19.

There should probably be proper tests for parsing the different kinds of paths, but I don't know how to write them.

A few things to note:
* I'm not sure what `Data.FilePath.Body.Sep` is for, as it wasn't used anywhere. I've made it platform dependent, and used it in the `Interpolation` implementation.
* Apparently, drive relative paths are a thing, e.g. `\some\path` which, if the current drive is `C:`, would be interpreted as `C:\some\path`. These are currently parsed as absolute Unix paths, but I've never seen them used so I'm not sure it's worth it to try to distinguish them somehow. (or if it's even necessary)
* Additionally, it appears there are "drive's current directory relative" paths, e.g. `D:some\path` (no `\` after the `:`), which is a path relative to the working directory on the `D:` drive, even if current drive is a different one. Apparently Windows shells track a separate working directory for each drive. Again, I've never seen this, so I haven't added support for it.

### CODE REVIEW

**Comment by stefan-hoeck** [comment]
Thank you so much for adding this. I can help with the tests, I think, bould could you first have a look into why CI fails?

And sorry for not replying earlier.

**Comment by elseLuna** [comment]
That should fix the lint.

The build is failing at `pack test filepath -n 1000`, so I booted up my linux partition and set pack to use the same nightly as in the CI (which installed all new packages, so it was effectively a fresh environment), and sure enough `pack test filepath` failed, and manually running `~/.pack/install/<commit>/idris2/bin/idris2 --build test/test.ipkg` gave the error:
```
Uncaught error: EmptyFC:Failed to resolve the dependencies for filepath-test:
  required hedgehog any but no matching version is installed
```
That was correct, `hedgehog` wasn't installed, and installing it fixed it.
However, that seemed unrelated so to be sure I removed `hedgehog` and reverted the pull request, and that also fixed it...?

So then I installed `pack` on a fresh VM I had lying around, and ran the same:
```
pack install filepath
pack test filepath -n 1000
```
And it worked fine, in fact `pack test filepath` installed `hedgehog` automatically, which I'm guessing is the intended behaviour.
I don't know why that wasn't happening on my linux partition.

So in conclusion: I have no idea what's going on.

**Comment by stefan-hoeck** [comment]
OK, thank. I'm going to merge this and will then have a look at the test runner. Thanks again for your contribution.
